src/app/modules/form/admin-form/__tests__/admin-form.payments.service.spec.ts

```diff
@@ -254,19 +258,157 @@ describe('admin-form.payment.service', () => {
+        ] as PaymentsProductUpdateDto
+
+        const putSpy = jest
+          .spyOn(EncryptFormModel, 'updatePaymentsProductById')
+          .mockResolvedValueOnce({
+            payments_field: {},
+          } as unknown as IEncryptedFormDocument)
+
+        // Act
+        const actualResult =
+          await AdminFormPaymentService.updatePaymentsProduct(
+            mockFormId,
+            updatedProducts,
+          )
+
+        expect(putSpy).toHaveBeenCalledOnce()
+        expect(actualResult.isOk()).toBeTrue()
+      })
+
+      it('should allow updates when product * max qty is at max amount', async () => {
+        jest.replaceProperty(PaymentConfig, 'paymentConfig', {
+          ...PaymentConfig.paymentConfig,
+          maxPaymentAmountCents: 100,
+        })
+        const updatedProducts = [
+          {
+            multi_qty: true,
+            max_qty: 10,
+            amount_cents: 10,
+          },
+        ] as PaymentsProductUpdateDto
+
+        const putSpy = jest
+          .spyOn(EncryptFormModel, 'updatePaymentsProductById')
+          .mockResolvedValueOnce({
+            payments_field: {},
+          } as unknown as IEncryptedFormDocument)
+
+        // Act
+        const actualResult =
+          await AdminFormPaymentService.updatePaymentsProduct(
+            mockFormId,
+            updatedProducts,
+          )
+
+        expect(putSpy).toHaveBeenCalled()
+        expect(actualResult.isOk()).toBeTrue()
+      })
+
+      it('should disallow updates when product * max qty exceeds max amount', async () => {
+        jest.replaceProperty(PaymentConfig, 'paymentConfig', {
+          ...PaymentConfig.paymentConfig,
+          maxPaymentAmountCents: 100,
+        })
+        const updatedProducts = [
+          {
+            multi_qty: true,
+            max_qty: 11,
+            amount_cents: 10,
+          },
+        ] as PaymentsProductUpdateDto
+
+        const putSpy = jest
+          .spyOn(EncryptFormModel, 'updatePaymentsProductById')
+          .mockResolvedValueOnce({
+            payments_field: {},
+          } as unknown as IEncryptedFormDocument)
+
+        // Act
+        const actualResult =
+          await AdminFormPaymentService.updatePaymentsProduct(
+            mockFormId,
+            updatedProducts,
+          )
+
+        expect(putSpy).not.toHaveBeenCalled()
+        expect(actualResult.isErr()).toBeTrue()
+        expect(actualResult._unsafeUnwrapErr()).toBeInstanceOf(
+          InvalidPaymentAmountError,
+        )
+      })
+    })
+
+    describe('with multi qty disabled', () => {
+      beforeEach(() => {
+        jest.clearAllMocks()
+      })
+      it('should allow updates when product * max qty possibly exceeds max amount', async () => {
+        jest.replaceProperty(PaymentConfig, 'paymentConfig', {
+          ...PaymentConfig.paymentConfig,
+          maxPaymentAmountCents: 100,
+        })
+        const updatedProducts = [
+          {
+            multi_qty: true,
```

Comment:
this should be `false` instead of `true` haha I think that's why the tests are failing too
```suggestion
            multi_qty: false,
```

Reply:
Great catch! Added in 6a477304f72877600a12753f5417d70ba6a990f0